core/src/main/java/tc/oc/pgm/stats/StatsMatchModule.java

```diff
@@ -0,0 +1,489 @@
+                  viewer.getInventory().setItem(7, createToolItem(viewer));
+                }
+              }
+            },
+            5 + 1, // NOTE: This is 1 second after the votebook appears
+            TimeUnit.SECONDS);
+
+    final Collection<Competitor> competitors = match.getCompetitors();
+
+    if (verboseStats && competitors.stream().allMatch(c -> c instanceof Team)) {
+
+      final List<InventoryMenuItem> items =
+          competitors.stream()
+              .map(c -> new TeamInventoryMenuItem(match, c))
+              .collect(Collectors.toList());
+
+      endOfMatchMenu =
+          competitors.size() <= 4
+              ? InventoryMenuUtils.smallMenu(match, verboseStatsTitle, items)
+              : InventoryMenuUtils.progressiveMenu(match, verboseStatsTitle, items);
+    }
+  }
+
+  private ItemStack createToolItem(MatchPlayer player) {
+    ItemStack tool = new ItemStack(Material.RED_SANDSTONE);
+    ItemMeta meta = tool.getItemMeta();
+    Component displayName =
+        TranslatableComponent.of("match.stats.title", TextColor.GREEN, TextDecoration.BOLD);
+    Component lore = TranslatableComponent.of("setting.lore", TextColor.GRAY);
+    meta.setDisplayName(TextTranslations.translateLegacy(displayName, player.getBukkit()));
+    meta.setLore(Lists.newArrayList(TextTranslations.translateLegacy(lore, player.getBukkit())));
+    meta.addItemFlags(ItemFlag.values());
+    tool.setItemMeta(meta);
+    return tool;
+  }
+
+  @EventHandler
+  public void onToolClick(PlayerInteractEvent event) {
+    if (!verboseStats) return;
+    Action action = event.getAction();
+    if ((action == Action.RIGHT_CLICK_AIR || action == Action.RIGHT_CLICK_BLOCK)
+        && match.isFinished()) {
+      ItemStack item = event.getPlayer().getItemInHand();
+
+      if (item.getType() == Material.RED_SANDSTONE) {
+        MatchPlayer player = match.getPlayer(event.getPlayer());
+        if (player == null) return;
+        endOfMatchMenu.display(player);
+      }
+    }
+  }
+
+  private Map.Entry<UUID, Integer> sortStats(Map<UUID, Integer> map) {
+    return map.entrySet().stream().max(Comparator.comparingInt(Map.Entry::getValue)).orElse(null);
+  }
+
+  private Map.Entry<UUID, Double> sortStatsDouble(Map<UUID, Double> map) {
+    return map.entrySet().stream()
+        .max(Comparator.comparingDouble(Map.Entry::getValue))
+        .orElse(null);
+  }
+
+  Component getMessage(
+      String messageKey, Map.Entry<UUID, ? extends Number> mapEntry, TextColor color) {
+    return TranslatableComponent.of(
+        messageKey, playerName(mapEntry.getKey()), numberComponent(mapEntry.getValue(), color));
+  }
+
+  public static Component numberComponent(Number stat, TextColor color) {
+    double doubleStat = stat.doubleValue();
+    if (doubleStat % 1 == 0) return TextComponent.of(stat.intValue(), color, TextDecoration.BOLD);
+    else return TextComponent.of(doubleStat, color, TextDecoration.BOLD);
```

Comment:
Use a number formatter here, to set the format you want (with 2 decimals) and pass the resulting formatted string to TextComponent